### ISSUE TITLE
Add Appsignal.configure besides YAML configuration

### ISSUE THREAD

**Issue by legendetm** [appsignal/appsignal-ruby#598]
Please add initializer support for Appsignal. 
```
Appsignal.configure do |config|
  config.active = ENV['APPSIGNAL_PUSH_API_KEY'].present?
  config.push_api_key = ENV['APPSIGNAL_PUSH_API_KEY']
  config.revision = Rails.root.join('REVISION').exist? ? Rails.root.join('REVISION').read : nil
  config.filter_parameters = Rails.application.config.filter_parameters.map(&:to_s)
  config.request_headers += %w[HTTP_X_CUSTOMER_HEADER]
  confif.ignore_errors += %w[...]
  ...
end
```

- Using AppSignal for Ruby gem version 2.10.4
- In my app using Rails version 6.0.2

**Comment by tombruijn** [comment]
Hi there, it's currently possible to configure AppSignal using Ruby code . Instead of a DSL you are able to pass along a hash to the `Appsignal::Config.new` method.

```ruby
Appsignal.config = Appsignal::Config.new(
  Rails.root,
  Rails.env,
  {
    :active => true
  }
)
```

Do note, that you will not have to do the check for the active conflict option. If the push API key is not present the config will not be active. https://docs.appsignal.com/ruby/configuration/options.html#option-active

**Comment by legendetm** [comment]
With that solution it's unclear for me, if I have to call `AppSignal.start` manually

**Comment by tombruijn** [comment]
Yes, you should call `Appsignal.start` manually, if in a Rails app,  in an initializer, so that it overwrites the original config loaded during Rails boot.

I understand the advantages, but we don't have plans to support this DSL at this time.

**Comment by legendetm** [comment]
Do I have to call any other methods manually like integerations (Sidekiq, Redis) or all other things are automatically done?

For example https://docs.appsignal.com/ruby/integrations/rack.html is called start_logger.

**Comment by tombruijn** [comment]
Our integration already calls start logger, it should not be necessary to call it unless you change any logger config options. Since our Railtie already calls it. There should not be any other methods you have to call, the rest is done automatically.

See there is no documentation page for this specific configuration method, I will create an issue on our documentation repository for this.
